workspace/utilities/lib/attr.xsl

```
@@ -5,291 +5,284 @@
 
-	<!-- Import untouched attribute back -->
-	<xsl:variable name="temp2">
-		<xsl:copy-of select="$temp" />
-
-		<xsl:for-each select="exsl:node-set($attrResult)/root/*" >
-			<xsl:variable name="attr-name" select="name()" />
-
-			<!-- Check if not present -->
-			<xsl:if test="count(exsl:node-set($temp)//*[name() = $attr-name]) = 0">
-				<xsl:copy-of select="." />
+				<!-- Output attribute -->
+				<xsl:attribute name="{$attr-name}">
+					<!-- Loop throught each values-->
+					<xsl:for-each select="*">
+						<!-- Add space before new attribute (except first) -->
+						<xsl:if test="position() &gt; 1">
+							<xsl:text> </xsl:text>
+						</xsl:if>
+						<!-- Output value -->
+						<xsl:value-of select="." />
+					</xsl:for-each>
+				</xsl:attribute>
 			</xsl:if>
 		</xsl:for-each>
-	</xsl:variable>
 
-	<xsl:variable name="newAttrResult">
-		<root>
-			<xsl:copy-of select="$temp2" />
-		</root>
-	</xsl:variable>
-
-	<xsl:choose>
-		<xsl:when test="count(./following-sibling::*) &gt; 0">
-			<!-- Recursive call -->
-			<xsl:apply-templates select="./following-sibling::*[1]" mode="node-attributes-rec">
-				<xsl:with-param name="attrResult" select="$newAttrResult" />
-			</xsl:apply-templates>
-		</xsl:when>
-		<xsl:otherwise>
-			<!-- Final render -->
-			<xsl:copy-of select="$newAttrResult" />
-		</xsl:otherwise>
-	</xsl:choose>
-	
-</xsl:template>
+		<!-- Debug info if enabled -->
+		<xsl:if test="$attr-mode = 'debug' and $debug = true()">
+			<script type="text/debug">
+				<src>
+					<xsl:copy-of select="$attr" />
+				</src>
+				<compiled>
+					<xsl:copy-of select="$compiledAttrXml" />
+				</compiled>
+			</script>
+		</xsl:if>
+	<!-- 																	/-->
+	</xsl:template>
 
+<!-- HELPER-TEMPLATES: COMPUTE ADD | SET | REM ==================================================-->
+	<xsl:template match="add | set" mode="node-attributes-rec">
+		<xsl:param name="attrResult" />
```

Can we throw an error if node other then add | set | rem are passed ?